### PR TITLE
Indexing fix in multiple python scripts

### PR DESCRIPTION
**Pull request by mpaiao**
Minor updates on multiple python scripts, for indexing and for using netcdf from scipy.io libraries

### Description:

I don't know if this is only happening on my computer, but `FatesPFTIndexSwapper.py ` started to issue the following indexing error when it attempted to copy scalars:
```
Creating Variable:  fates_canopy_closure_thresh
Traceback (most recent call last):
  File "/path_to/E3SM/components/elm/src/external_models/fates/tools/FatesPFTIndexSwapper.py", line 296, in <module>
    main(sys.argv)
    ~~~~^^^^^^^^^^
  File "/path_to/Models/E3SM/components/elm/src/external_models/fates/tools/FatesPFTIndexSwapper.py", line 207, in main
    out_var.assignValue(float(fp_in.variables.get(key).data))
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/scipy/io/_netcdf.py", line 943, in assignValue
    self.data[:] = value
    ~~~~~~~~~^^^
IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed
<sys>:0: RuntimeWarning: Cannot close a netcdf_file opened with mmap=True, when netcdf_variables or arrays referring to its data still exist. All data arrays obtained from such files refer directly to data on disk, and must be copied before the file can be cleanly closed. (See netcdf_file docstring for more information on mmap.)

```
The suggested change in the code seems to work fine (though I still get the RuntimeWarning message). I made similar updates in a few other scripts. I also updated `ncvarsort.py` to use scipy.

### Expectation of Answer Changes:

No change in the FATES code, just the tool.

### Checklist

*If this is your first time contributing, please read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/main/CONTRIBUTING.md) document.*

All checklist items must be checked to enable merging this pull request:

*Contributor*
- [ ] The in-code documentation has been updated with descriptive comments
- [ ] The documentation has been assessed to determine if updates are necessary

*Integrator*
- [ ] FATES PASS/FAIL regression tests were run
- [ ] Evaluation of test results for answer changes was performed and results provided

### Documentation

- [Technical Note](https://github.com/NGEET/fates-docs) update:
- [User's Guide](https://github.com/NGEET/fates-users-guide) update: 

### Test Results:

*CTSM (or) E3SM (specify which) test hash-tag:*

*CTSM (or) E3SM (specify which) baseline hash-tag:*

*FATES baseline hash-tag:*

*Test Output:*

### CODE REVIEW

File: tools/ncvarsort.py [hunk 1]
```diff
@@ -4,11 +4,19 @@
 # --input or --fin: input filename.
 # --output or --fout: output filename.  If missing, will assume its directly modifying the input file, and will prompt unless -O is specified
 
-import netCDF4 as nc
+#import netCDF4 as nc
```
Comment (rgknox) [inline tools/ncvarsort.py:7]: nice catch

File: tools/ncvarsort.py [hunk 2]
```diff
@@ -118,16 +126,31 @@ def main():
     # as well as all metadata to the new file.
     for i in range(len(varnames_list_sorted)):
         v_name = varnames_list_sorted[i]
-        varin = dsin.variables[v_name]
-        outVar = dsout.createVariable(v_name, varin.datatype, varin.dimensions)
+        varin  = dsin.variables.get(v_name)
+        v_type = dsin.variables[v_name].typecode()
+        v_dims = varin.dimensions
+        print(" V_NAME = ",v_name,"; V_TYPE = ",v_type,"; V_DIMS = ",v_dims)
```
Comment (rgknox) [inline tools/ncvarsort.py:132]: do you want to keep this print statement, or was it just for debugging?

Comment (mpaiao) [reply tools/ncvarsort.py:132]: It was just for debugging, I will delete it.

**Comment by rgknox** [comment]
These fixes were necessary to get UpdateParamAPI.py working (which calls a few of these routines). UpdateParamAPI.py will be needed to advance our parameter file API changes coming down the line.  Lets prioritize this if possible.

**Review by rgknox — APPROVED** [review]

**Pull request merged**